docs/cli/orchestration/scripts.md

```diff
@@ -0,0 +1,180 @@
+inherited the script definition and that are reachable from the current working directory and not affected by any filters.
+Scripts can be redefined on any level of the hierarchy and replace the previous definition.
+
+::: info
+**Note:** No error or warning will be issued for stacks not selected for script execution.
+:::
+
+A more detailed explanation is covered in the Scope and Visibility sections where:
+
+- Scope defines where a script will be executed in
+- Visibility describes where a script can be called from
+
+### Scope
+
+In general, the scope of a script definition includes the directory it is defined in and all nested sub-directories.
+A script can be run on stacks within its scope.
+
+**Figure A** shows an example by highlighting the scope of a script named `cmd`:
+
+<img src="../assets/orchestration/scripts-scope.svg" width="600px" alt="Shows an example by highlighting the scope of a script named" />
+
+Which stacks are selected further depends on additional criteria. Most importantly, the directory  `terramate script run`
+is executed from, limits the selected stacks to those contained within. This works just like `terramate run`.
+
+A useful property of scripts is that they allow multiple definitions with the same name at different locations, yet all
+can be run with a single `terramate script run` invocation. For example, as shown in in **Figure B**,
+`terramate script run cmd` at the root level will run each of the two script definitions on stacks within their respective scopes.
+
+As shown in **Figure C**, a script that is re-defined in a sub-directory overriding its the previous definition will only
+be executed in stacks contained within the sub-directory, while the other stacks execute the original definition of the script.
+
+This behavior enables scenarios where a single `terramate script run deploy` could execute different scripts for
+different parts of the infrastructure, based on directory grouping.
+
+### Visibility
+
+To be able to call a script, it needs to be visible to the current working directory.
+Script are generally visible within their scope, as defined in the last section, but additionally they are also visible
+to all their parent directories. This enables us to call a script from the project root without knowing where it is
+actually defined as also shown in **Figure D**.
+
+<img src="../assets/orchestration/scripts-visibility.svg" width="600px" alt="Explaining the visibility of a script in Terramate" />
+
+## Inspecting Scripts
+
+To inspect the behavior of a script, a dry-run can be started with `script run --dry-run <command...>`.
+This will print out the commands that would be executed per stack, without actually executing them.
+
+Besides this, there are additional experimental commands that can be used to inspect and debug script definitions:
+
+- [`terramate experimental script list`]()
```

Comment:
What's the reason for adding an empty hyperlink here? 🤔 
This renders like this (and the links are pointing to self i.e.` /cli/orchestration/`)

![image](https://github.com/terramate-io/terramate/assets/92498/b682e951-e93e-4a6a-a9e6-1dc917cf89a9)

Reply:
good catch will fix this!

Reply:
@wmalik maybe it makes sense to always the "The `script tree` command" and adding a link to the actual command instead of writing the full command in here?

Reply:
That makes sense. By the way, these commands have already been moved to the top level (they are no longer plunder experimental)

Reply:
sweet thank you